async-coap-uri/src/escape/unescape_uri.rs

```diff
@@ -311,16 +406,20 @@ impl<'a> Iterator for UnescapeUri<'a> {
                         // This is a UTF8 byte.
                         utf8_buf[utf8_len] = decoded;
                         utf8_len += 1;
+
                         if utf8_len >= UTF8_CHAR_WIDTH[utf8_buf[0] as usize] as usize {
                             if let Ok(utf8_str) = from_utf8(&utf8_buf[..utf8_len]) {
                                 return Some(utf8_str.chars().next().unwrap());
                             } else {
-                                self.had_error = true;
+                                self.had_error = Some(UnescapeError::InvalidUtf8 {
+                                    buf: utf8_buf,
+                                    len: utf8_len,
+                                });
```

Comment:
This is one of the errors that I am not sure about

Reply:
This error happens when we have received all of the UTF8 bytes that were indicated by the first byte, but the resulting sequence doesn't decode properly into a valid UTF character. For example, `%f0%9f%90%f0` contains four UTF8 byte codes (they all have the MSB set), but together they do not decode properly.